### PR TITLE
Loss bugfix and quality of life

### PR DESCRIPTION
**Pull request by klemenvon**
Fixed issue where loss would sometimes come out as float rather than a torch.Tensor
Added ability to glob training data recursively, so we find more/all of it.

### CODE REVIEW

**Review by SarodYatawatta — APPROVED** [review]
Nice work
